v2/blockstore/readonly.go

```diff
@@ -320,6 +348,16 @@ func (b *ReadOnly) GetSize(key cid.Cid) (int, error) {
 	return fnSize, nil
 }
 
+func isIdentity(key cid.Cid) (ok bool, digest []byte, err error) {
+	dmh, err := multihash.Decode(key.Hash())
+	if err != nil {
+		return
+	}
+	ok = dmh.Code == multihash.IDENTITY
+	digest = dmh.Digest
+	return
```

Comment:
best to not use naked returns if you can avoid it: https://github.com/golang/go/wiki/CodeReviewComments#naked-returns

I personally think they're only OK with the defer-error-check pattern, as you really need them in that case